packages/core-sdk/src/resources/dispute.ts

```diff
@@ -19,102 +22,240 @@ export class DisputeClient {
     this.rpcClient = rpcClient;
     this.wallet = wallet;
   }
-  /*
-  public async whitelistDisputeTags(
-    request: WhitelistDisputeTagsRequest,
-  ): Promise<WhitelistDisputeTagsResponse> {
+
+  /**
+   * Initiates a dispute on an IP.
+   * @param request - The request object containing necessary data to raise a dispute.
+   *   @param request.targetIpId - The IP ID that is the target of the dispute.
+   *   @param request.arbitrationPolicy - The address of the arbitration policy.
+   *   @param request.linkToDisputeEvidence - The link to the dispute evidence.
+   *   @param request.targetTag - The target tag of the dispute.
+   *   @param request.calldata - Optional calldata to initialize the policy.
+   *   @param request.txOptions - Optional transaction options.
+   * @returns A Promise that resolves to a RaiseDisputeResponse containing the transaction hash.
+   * @throws `NotRegisteredIpId` if targetIpId is not registered in the IPA Registry.
+   * @throws `NotWhitelistedDisputeTag` if targetTag is not whitelisted.
+   * @throws `ZeroLinkToDisputeEvidence` if linkToDisputeEvidence is empty
+   * @calls raiseDispute(address _targetIpId, string memory _linkToDisputeEvidence, bytes32 _targetTag, bytes calldata _data) external nonReentrant returns (uint256) {
+   * @emits DisputeRaised (disputeId_, targetIpId, msg.sender, arbitrationPolicy, linkToDisputeEvidence, targetTag, calldata);
+   */
+  public async raiseDispute(request: RaiseDisputeRequest): Promise<RaiseDisputeResponse> {
     try {
       const { request: call } = await this.rpcClient.simulateContract({
         ...DisputeModuleConfig,
-        functionName: "whitelistDisputeTags",
-        args: [stringToHex(request.tag, { size: 32 }), request.allowed],
+        functionName: "raiseDispute",
+        args: [
+          request.targetIpId,
+          request.linkToDisputeEvidence,
+          stringToHex(request.targetTag, { size: 32 }),
+          request.calldata ? request.calldata : "0x",
+        ],
       });
 
       const txHash = await this.wallet.writeContract(call);
 
+      if (request.txOptions?.waitForTransaction) {
+        await waitTx(this.rpcClient, txHash);
```

Comment:
I think we should return dispute id in the response, so that users can use dispute id to query API to get dispute information